docs: reorder sdk docs

vinckr opened the pull request:
<!--
Describe the big picture of your changes here to communicate to the maintainers why we should accept this pull request.

This text will be included in the changelog. If applicable, include links to documentation or pieces of code.
If your change includes breaking changes please add a code block documenting the breaking change:

```
BREAKING CHANGES: This patch changes the behavior of configuration item `foo` to do bar. To keep the existing
behavior please do baz.
```
-->

## Related Issue or Design Document

<!--
If this pull request

1. is a fix for a known bug, link the issue where the bug was reported in the format of `#1234`;
2. is a fix for a previously unknown bug, explain the bug and how to reproduce it in this pull request;
3. implements a new feature, link the issue containing the design document in the format of `#1234`;
4. improves the documentation, no issue reference is required.

Pull requests introducing new features, which do not have a design document linked are more likely to be rejected and take on average 2-8 weeks longer to
get merged.

You can discuss changes with maintainers either in the Github Discussions in this repository or
join the [Ory Chat](https://www.ory.sh/chat).
-->

## Checklist

<!--
Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of
them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code.
-->

- [x] I have read the [contributing guidelines](../blob/master/CONTRIBUTING.md) and signed the CLA.
- [x] I have referenced an issue containing the design document if my change introduces a new feature.
- [x] I have read the [security policy](../security/policy).
- [x] I confirm that this pull request does not address a security vulnerability. 
      If this pull request addresses a security vulnerability, 
      I confirm that I got approval (please contact [security@ory.sh](mailto:security@ory.sh)) from the maintainers to push the changes.
- [x] I have added tests that prove my fix is effective or that my feature works.
- [x] I have added the necessary documentation within the code base (if appropriate).

## Further comments

<!--
If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution
you did and what alternatives you considered, etc...
-->

aeneasr commented on docs/sdk.mdx:
```diff
@@ -0,0 +1,58 @@
+
+:::
+
+:::caution old sdks
+
+[`hydra-client`](https://github.com/ory/sdk/tree/master/clients/hydra),
+[`kratos-client`](https://github.com/ory/sdk/tree/master/clients/kratos), and
+[`keto-client`](https://github.com/ory/sdk/tree/master/clients/keto) SDKs are no longer actively maintained.  
+Read the [Ory SDK upgrade guide](https://www.ory.sh/docs/guides/upgrade/sdk-v1) to learn more.
+
+:::
```
I think we'll keep these SDKs around for a while

```suggestion
:::caution Self-hosted SDKs

Ory releases SDKs which can more easily be used when self-hosting Ory software. These SDKs follow the open source release schedule. These SDKs are available for [`hydra-client`](https://github.com/ory/sdk/tree/master/clients/hydra),
[`kratos-client`](https://github.com/ory/sdk/tree/master/clients/kratos), and
[`keto-client`](https://github.com/ory/sdk/tree/master/clients/keto).

:::
```

vinckr replied:
So it is recommended to use the "old" sdk when selfhosting? 

bc earlier it says

>It (the Ory "client" SDK) works with Ory Network and when self-hosting Ory services.

aeneasr replied:
It's a bit ambigous. Both SDKs work with both systems. But the ory client SDK follows the release steps of the Ory Network, and the other SDKs follow the release steps of the self-hosted stuff. That's the difference. So if you use ory client v1.2.3 then you don't know what version of Ory Kratos that is compatible with, for example.

vinckr replied:
Gotcha, 
then I would definitely recommend to use the kratos-client etc. SDK when selfhosting
and you have to use client for Ory Network (pointing out the difference)
and have a dedicated SDK page for kratos etc. in the selfhosting reference section.

putting this back in draft state until I have done the above and ask for review again in a bit.

vinckr commented:
changed it so it has a main SDK doc and one for each project. 
Also new format + text, and added a note that hopefully explains which SDK to use when.

tomekpapiernik commented on docs/hydra/sdk/01_overview.md:
```diff
@@ -1,40 +1,62 @@
 
-Before you check out the SDKs, head over to the [REST API](reference/api.mdx) documentation which includes code samples for common
-programming languages for each REST endpoint.
+Before using the SDK, it is recommended to consult the [Ory REST API](../reference/api) documentation, which includes code samples
```
Please use active voice and imperative mood. 
The way we're linking to docs now must use filenames, so please revert to `api.mdx`.
Also, are there actual code samples/examples for various programming languages in the API reference?

vinckr replied:
done, 

as for the code samples in some cases there is someting like 

```js 
pseudo-code example 
router.get('/recovery', async function (req, res) { const flow = await client.getVerificationFlow(req.header('cookie'), req.query['flow'])
res.render('verification', flow) })
```

It does contain explanations on every REST endpoint. let me know if you can think of a better phrasing

tomekpapiernik reviewed (changes requested):
The docs you edited have a similar structure so I didn't duplicate my comments. Make sure to address the highlighted issues in all docs in this PR. Thanks!

vinckr commented on src/sidebar.js:
```diff
@@ -609,30 +608,27 @@ module.exports = {
-          type: "autogenerated",
-          dirName: "keto/sdk",
+          type: "category",
+          label: "Ory OAuth2",
```
you mean Ory OAuth2 & OpenID, right?

vinckr commented:
Hey @tomekpapiernik 
I changed the formatting, so only one admonition left. 
Let me know if you can think of a better heading than "Further information". 
Addressed your other comments as well :) 
Basically 3 things to be conveyed in that section:

- the link to the source code
- the call to action to create an issue if you are missing your language
- the note which SDK to use for Ory self-hosted & Network, vice versa

also the Hydra SDK has a some extra info on the oauth2 endpoints.

aeneasr commented:
This looks good so far to me! The SDK examples themselves need to be reworked, but we can do that in another PR :)
